### ISSUE TITLE
Unable to Get JSON Data With WP Rest API V2

### ISSUE THREAD

**Issue by lekeope** [WP-API/WP-API#3005]
in Version 1 when i used `/wp-json/posts`  i get a list of post in JSON format but since i updated to version 2 (With Wordpress Version 4.7.2) when i use `/wp-json/wp/v2/posts`  i don't get any data in JSON data i am only redirected to the homepage of the website.

What could be Wrong??

**Comment by nylen** [comment]
The REST API has been merged into WordPress core! Development and support are no longer taking place in this repository.

Please post support requests to the WordPress forums at https://wordpress.org/support/forum/how-to-and-troubleshooting/ with the topic tag "rest-api".

For bugs and patches, please post the issue or patch to WordPress core Trac at https://core.trac.wordpress.org -- Be sure to include full details and reproduction steps about the issue you are experiencing.

If you are unfamiliar with Trac, you can refer to the "Opening a Ticket" user guide at https://make.wordpress.org/core/handbook/tutorials/trac/opening-a-ticket/ for an introduction!

Thank you,
The REST API team

**Comment by kadamwhite** [comment]
@Edge-Developer `/wp-json/wp/v2/posts` should work with 4.7.2; you no longer need the plugin installed. I'd suggest uninstalling the plugin, flushing your permalinks (by visiting your permalink settings page and hitting "save"), and trying again; if you continue having difficulty, please open a forum post as directed above.

**Comment by lekeope** [comment]
@kadamwhite Thanks Works Like Charm. Thanks!!!

**Comment by lekeope** [comment]
@kadamwhite Thanks once again. I had given up on the solution.

Please how do i secure/authenticate the API, so that i alone can have access to the JSON the data

**Comment by kadamwhite** [comment]
@edge-developer glad to hear that worked! Check out the authentication and FAQ section on http://developer.wordpress.org/rest-api for more information about securing the API.

**Comment by lekeope** [comment]
@kadamwhite Thanks Again

**Comment by lekeope** [comment]
@kadamwhite I checked out the WP Rest API version 2, although it has some new cool features ( such as comments, post offset etc) but it i still prefer the version 1 for now. For instance in version 2, thumbnail url/details are to be called separately which is not the case in version 1 (Everything: thumbnail, content, title are entirely available with /wp-json/posts). In Version 2 Post "Tags" are nowhere to be found.

Can i edit the API(if so how, PS: documentation doesn't state how) or we would have to wait on wordpress to make amends?

Thanks

**Comment by kadamwhite** [comment]
@edge-developer please read through the documentation for the core REST API (at http://developer.wordpress.org/rest-api); the issue with tags and images coming back as separate responses is handled by passing the `?_embed` parameter with your request, for example. We spent a long time improving the performance, security and flexibility of the API since version 1 and I think you'll find that the built in REST API that comes with WP 4.7 is far more powerful than the old code. Best of luck with your projects

**Comment by lekeope** [comment]
Thanks @kadamwhite

**Comment by lekeope** [comment]
How do i get it to include the tag? here's what is tried `/wp-json/wp/v2/posts?_embed=tags` . I didn't get the tags.

and how do i remove unnecessary info (it only bloats my JSON :(

**Comment by lekeope** [comment]
and how can i include the thumbnail url too. Thanks for the Support @kadamwhite

**Comment by kadamwhite** [comment]
@edge-developer please refer to the [documentation on `?_embed`](https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/), the tags will be exposed within the `wp:term` key in the `_embedded` property on the response object. The thumbnail attachment info will also be present within `_embedded`.

There is not currently any native way to filter the fields returned from the API but several plugins exist to do this, search for rest filter fields to get started.

As posted above we've moved support to the forums; If you have further support questions please post them at https://wordpress.org/support/forum/how-to-and-troubleshooting/ with the topic tag "rest-api". Thanks!
